### ISSUE TITLE
The prediction with use_best_model=True after early_stopping match neither best nor last iteration

### ISSUE THREAD

**Issue by Edvard88** [catboost/catboost#604]
My steps were:
1. Fit the model with early_stopping and use_best_model=True and make the prediction on X_val (test set)
2. Fit the model with n_estimators=best_iteration + 1, early_stopping=False and make the prediction on  X_val
3. Fit the model with n_estimators=last_iteration + 1, early_stopping=False and make the prediction on  X_val
4. Compare prediction from step 1 with step 2 and 3.

The issue is:  the prediction after early_stopping (step1) 
doesn't match with predictions of the best (step2) or the last model (step3). 
Seems that fit with  use_best_model=True  after early_stopping should return the best model. 

Catboost version: '0.11.2'

**Comment by annaveronika** [comment]
If you use binary classification and do not specify learning_rate, then learning_rate is selected by default and depends on number of iterations. Is this the case?

**Comment by Edvard88** [comment]
I use binary classification and specify learning_rate=0.1
 
`estimator = cat.CatBoostClassifier(n_estimators=1000, use_best_model=True ,eval_metric='AUC', learning_rate=0.1, random_state=42)
`

PS In steps 2 and 3 I've tried n_estimators=best(last)_iteration + 1 and  just n_estimators=best(last)_iteration, without +1.   The result was the same

**Comment by Edvard88** [comment]
@annaveronika really if fix learning_rate everything will be fine.

**Comment by Edvard88** [comment]
PS But interesting. The documentation says that learning_rate : float, [default=0.03]

**Comment by annaveronika** [comment]
Sorry for that, we didn't update the docstring, it's fixed now in code, will be on pypi in the next release.
Closing the issue then. Thanks a lot for the report!
